python/graphstorm/trainer/lp_trainer.py

```diff
@@ -121,34 +122,43 @@ def fit(self, train_loader, num_epochs,
                 self._model.unfreeze_input_encoder()
             # TODO(xiangsx) Support unfreezing gnn encoder and decoder
 
+            rt_profiler.record('train_init')
             for i, (input_nodes, pos_graph, neg_graph, blocks) in enumerate(train_loader):
+                rt_profiler.record('train_sample')
```

Comment:
If we do not use th.distributed.sync, the profiling result may be less accurate due to load imbalance.

Reply:
the analyzer tries to compute the variance in each step as well.

Reply:
we cannot put barriers in each step.

Reply:
Barrier can be inside the profiler.

Reply:
done